src/meta/src/hummock/manager/mod.rs

```diff
@@ -2736,16 +2736,7 @@ async fn write_exclusive_cluster_id(
 
     let cluster_id_dir = format!("{}/{}/", state_store_dir, CLUSTER_ID_DIR);
     let cluster_id_full_path = format!("{}{}", cluster_id_dir, CLUSTER_ID_NAME);
-    let metadata = match object_store.list(&cluster_id_dir).await {
-        Ok(metadata) => metadata,
-        Err(_) => {
-            return Err(ObjectError::internal(
-                "Fail to access remote object storage, 
```

Comment:
Here we want to see this error message every time meta fails to access object storage, and not only aws s3, but also gcs, oss...
So maybe we can not limite it to only aws error.

Reply:
It's not limited to AWS error. Here we just propagate the error, so the message is completely decided by the error itself.

The problem is AWS error's `Display`, which is fixed by this PR.

e.g., this is the panic msg of gcs
```
thread 'risingwave-main' panicked at 'called `Result::unwrap()` on an `Err` value: ObjectStore(Unexpected (temporary) at Pager::next, context: { called: reqsign::LoadCredential, service: gcs, path: hummock_001/cluster_id/ } => loading credential to sign http request, source: error sending request for url (http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=https://www.googleapis.com/auth/devstorage.read_write): error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
  backtrace of `ObjectError`:

```

BTW, opendal even has other logging
```
2023-07-28T12:56:18.140486+02:00 ERROR opendal::services: service=gcs operation=list path=hummock_001/cluster_id/ -> failed: Unexpected (temporary) at Pager::next => loading credential to sign http request

Context:
    called: reqsign::LoadCredential
    service: gcs
    path: hummock_001/cluster_id/

Source: error sending request for url (http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=https://www.googleapis.com/auth/devstorage.read_write): error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known

Caused by:
    0: error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
    1: dns error: failed to lookup address information: nodename nor servname provided, or not known
    2: failed to lookup address information: nodename nor servname provided, or not known

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
```

Reply:
I'm also comfortable to add more error message here, or panic directly. But we should at least also provide the original error message, which turns out to be very informative.

And personally I think "please check if your Access Key and Secret Key are configured correctly" is not always true and can be misleading, so I removed it.